internal/provider/application_resource.go

```diff
@@ -281,10 +282,81 @@ func (r *applicationResource) Delete(ctx context.Context, req resource.DeleteReq
 
 	err := r.client.DeleteApplication(state.ID.ValueString())
 	if err != nil {
-		resp.Diagnostics.AddError(
-			"Error Deleting Application",
-			"Could not delete Application ID "+state.ID.ValueString()+": "+err.Error(),
-		)
+		// Just take the error out for now, at least until we try below...
+		// resp.Diagnostics.AddError(
+		// 	"Error Deleting Application",
+		// 	"Could not delete Application ID "+state.ID.ValueString()+": "+err.Error(),
+		// )
+
+		// TODO: determine the error code here to kick in the next bit of logic.
+		// Just assume whatever the error, lets update the app anyway to remove CA Templates.
+		owners := []tlspc.OwnerAndType{}
+		for _, v := range state.Owners {
+			m := v.Elements()
+			// TODO: Work out how you're supposed to get an unquoted string out
+			kind := strings.Trim(m["type"].String(), `"`)
+			ownerId := strings.Trim(m["owner"].String(), `"`)
+			if kind != "USER" && kind != "TEAM" {
+				resp.Diagnostics.AddError(
+					"Error creating application",
+					"Could not create application, unsupported owner type: "+kind,
+				)
+				return
+			}
+			if ownerId == "" {
+				resp.Diagnostics.AddError(
+					"Error creating application",
+					"Could not create application, undefined owner",
+				)
+				return
+			}
+			owner := tlspc.OwnerAndType{
+				ID:   ownerId,
+				Type: kind,
+			}
+			owners = append(owners, owner)
+		}
+
+		// Set this as: {"":""} to use to overwrite the state
+		aliases := map[string]string{}
+
+		// for k, v := range state.CATemplateAliases.Elements() {
+		// 	aliases[k] = strings.Trim(v.String(), `"`)
+		// }
+
+		application := tlspc.Application{
+			ID:                   state.ID.ValueString(),
+			Name:                 state.Name.ValueString(),
+			Owners:               owners,
+			CertificateTemplates: aliases,
+		}
+
+		updated, err := r.client.UpdateApplication(application)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error updating application",
+				"Could not update application, unexpected error: "+err.Error(),
+			)
+			return
+		}
+		plan.ID = types.StringValue(updated.ID)
+		diags = resp.State.Set(ctx, plan)
```

Comment:
I'd hazard a guess that your problem is here, you don't have a fully instantiated `plan` - just an otherwise empty `applicationResourceModel` with the `ID` attribute set, and so it might be failing whatever schema validation checks that are in `State.Set`.